diff --git a/pixman/pixman-access.c b/pixman/pixman-access.c
index 4d91fb3..a1c203a 100644
--- a/pixman/pixman-access.c
+++ b/pixman/pixman-access.c
@@ -429,0 +430 @@ fetch_and_convert_pixel (bits_image_t *image,
+	(void)mask;							\
@@ -1749 +1750 @@ static const format_info_t accessors[] =
-    { PIXMAN_null },
+    { PIXMAN_null, NULL, NULL, NULL, NULL, NULL, NULL },
diff --git a/pixman/pixman-combine32.c b/pixman/pixman-combine32.c
index 78c3f15..b8b53fd 100644
--- a/pixman/pixman-combine32.c
+++ b/pixman/pixman-combine32.c
@@ -51 +51 @@ combine_mask_ca (uint32_t *src, uint32_t *mask)
-    if (a == ~0)
+    if (a == UINT32_MAX)
@@ -80 +80 @@ combine_mask_value_ca (uint32_t *src, const uint32_t *mask)
-    if (a == ~0)
+    if (a == UINT32_MAX)
@@ -101 +101 @@ combine_mask_alpha_ca (const uint32_t *src, uint32_t *mask)
-    if (a == ~0)
+    if (a == UINT32_MAX)
@@ -951 +951 @@ combine_in_reverse_ca (pixman_implementation_t *imp,
-	if (a != ~0)
+	if (a != UINT32_MAX)
@@ -1016 +1016 @@ combine_out_reverse_ca (pixman_implementation_t *imp,
-	if (a != ~0)
+	if (a != UINT32_MAX)
diff --git a/pixman/pixman-private.h b/pixman/pixman-private.h
index 97f8391..81d410d 100644
--- a/pixman/pixman-private.h
+++ b/pixman/pixman-private.h
@@ -849 +849 @@ PIXMAN_EXPORT int
-_pixman_implementation_get_reference_fast_path_size ();
+_pixman_implementation_get_reference_fast_path_size (void);
diff --git a/pixman/pixman-glyph.c b/pixman/pixman-glyph.c
index f6133b1..5597683 100644
--- a/pixman/pixman-glyph.c
+++ b/pixman/pixman-glyph.c
@@ -637 +637,2 @@ out:
-#if defined(__GNUC__) && !defined(__x86_64__) && !defined(__amd64__)
+#if defined(__GNUC__) && defined(__i386__) && !defined(__x86_64__) &&          \
+    !defined(__amd64__)
